package.json

```diff
@@ -12,18 +12,26 @@
     "@tiptap/pm": "^2.1.16",
     "@tiptap/react": "^2.1.16",
     "@tiptap/starter-kit": "^2.1.16",
+    "@types/jest": "^29.5.12",
+    "@types/node": "^20.11.19",
+    "@types/react": "^18.2.57",
+    "@types/react-dom": "^18.2.19",
     "bootstrap": "^5.3.2",
+    "immer": "^10.0.3",
     "interweave": "^13.1.0",
     "parchment": "^2.0.1",
     "react": "^18.2.0",
     "react-bootstrap": "^2.9.2",
     "react-d3-cloud": "^1.0.6",
     "react-dom": "^18.2.0",
+    "react-hook-form": "^7.50.1",
```

Comment:
I find react-hook-form a very good module but for your purposes right now it probably is overkill. I assumed you were doing more validation when I started on the submit page but how you have it setup right now with onChange being how update happens it's not adding a lot of value besides for Links where it's obfuscating away a few things. I don't know your opinion on using third party libraries and how much you like keeping control over vs abstractions for ease of use. 

Well you use bootstrap so you don't mind 3rd party completely. :P